third_party/hexagon/fully_connected.cc

```diff
@@ -52,92 +52,79 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
 #include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/hexagon/hexagon_fully_connected.h"
 #include "tensorflow/lite/micro/kernels/kernel_util.h"
-#include "third_party/hexagon/hexagon_tflm_translation_fully_connected.h"
+
+#include "hexagon_tflm_translation_fully_connected.h"
 
 namespace tflite {
 namespace {
 
-// Input tensors.
-constexpr int kInputTensor = 0;
-constexpr int kWeightsTensor = 1;
-constexpr int kBiasTensor = 2;
-// Output tensor.
-constexpr int kOutputTensor = 0;
-
-struct OpData {
-  // The scaling factor from input to output (aka the 'real multiplier') can
-  // be represented as a fixed point multiplier plus a left shift.
-  int32_t output_multiplier;
-  int output_shift;
-  // The range of the fused activation layer. For example for kNone and
-  // uint8_t these would be 0 and 255.
-  int32_t output_activation_min;
-  int32_t output_activation_max;
-  // The index of the temporary tensor where the quantized inputs are cached.
-  int input_quantized_index;
-  // Cached zero point values of tensors.
-  int32_t input_zero_point;
-  int32_t filter_zero_point;
-  int32_t output_zero_point;
-
-  void* hexagon_data;
-};
-
 TfLiteStatus CalculateOpData(TfLiteContext* context,
                              TfLiteFusedActivation activation,
                              TfLiteType data_type, const TfLiteTensor* input,
                              const TfLiteTensor* filter,
                              const TfLiteTensor* bias, TfLiteTensor* output,
-                             OpData* data) {
+                             HexagonOpDataFullyConnected* data) {
   TfLiteStatus status = kTfLiteOk;
   if (data_type != kTfLiteFloat32) {
     double real_multiplier = 0.0;
     TF_LITE_ENSURE_STATUS(GetQuantizedConvolutionMultipler(
         context, input, filter, bias, output, &real_multiplier));
     int exponent;
-    QuantizeMultiplier(real_multiplier, &data->output_multiplier, &exponent);
-    data->output_shift = -exponent;
+    QuantizeMultiplier(real_multiplier,
+                       &data->reference_op_data.output_multiplier, &exponent);
+    data->reference_op_data.output_shift = -exponent;
     TF_LITE_ENSURE_STATUS(CalculateActivationRangeQuantized(
-        context, activation, output, &data->output_activation_min,
-        &data->output_activation_max));
+        context, activation, output,
+        &data->reference_op_data.output_activation_min,
+        &data->reference_op_data.output_activation_max));
 
-    data->input_zero_point = input->params.zero_point;
-    data->filter_zero_point = filter->params.zero_point;
-    data->output_zero_point = output->params.zero_point;
+    data->reference_op_data.input_zero_point = input->params.zero_point;
+    data->reference_op_data.filter_zero_point = filter->params.zero_point;
+    data->reference_op_data.output_zero_point = output->params.zero_point;
   }
   return status;
 }
```

Comment:
use the common function from fully_comnnected.h?

https://github.com/tensorflow/tflite-micro/blob/4e091a46e935afcb27035e538044e37892d4a983/tensorflow/lite/micro/kernels/fully_connected_common.cc#L55-L59